File: leetcode/python/2870_minimum_number_of_operations_to_make_array_empty.py
from collections import Counter
from math import ceil


class Solution:
    def minOperations(self, nums: list[int]) -> int:
        count = Counter(nums)
        
        res = 0
        for _, v in count.items():
            if v == 1:
                return -1
            res += ceil(v / 3)
        return res
        
        

File: leetcode/rust/src/2870_minimum_number_of_operations_to_make_array_empty.rs
use std::collections::HashMap;

impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut count = HashMap::new();
        for n in nums {
            count.entry(n).and_modify(|e| *e += 1).or_insert(1);
        }

        let mut res = 0;
        for v in count.values() {
            if *v == 1 {
                return -1;
            }
            res += (*v as f32 / 3.0).ceil() as i32;
        }
        res
    }
}
// 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16
// x, 1, 1, 2, 2, 2, 3, 3, 3, 3,  4,  4,  5, 5, 5, 6